api/src/test/java/org/pmiops/workbench/api/WorkspacesControllerTest.java

```diff
@@ -917,4 +924,35 @@ public void testUnableToShareWithNonExistentUser() throws Exception {
     shareWorkspaceRequest.addItemsItem(writer);
     workspacesController.shareWorkspace(workspace.getNamespace(), workspace.getName(), shareWorkspaceRequest);
   }
+
+  @Test
+  public void testNoteBookList() throws Exception {
+    org.pmiops.workbench.firecloud.model.WorkspaceResponse fcResponse =
+        new org.pmiops.workbench.firecloud.model.WorkspaceResponse();
+    org.pmiops.workbench.firecloud.model.Workspace mockWorkspace = new org.pmiops.workbench.firecloud.model.Workspace();
+    mockWorkspace.setBucketName("MockBucketName");
+    fcResponse.setWorkspace(mockWorkspace);
+    when(fireCloudService.getWorkspace("mockProjectName", "mockWorkspaceName")).thenReturn(
+        fcResponse
+    );
+    List<String> blobDetailList = new ArrayList<String>();
+    String mockBlobDetail1 = new String("notebook/File.ipynb");
+    String mockBlobDetail2 = new String("notebook/File1.txt");
+    String mockBlobDetail3 = new String("config/File1.txt");
+    blobDetailList.add(mockBlobDetail1);
+    blobDetailList.add(mockBlobDetail2);
+    blobDetailList.add(mockBlobDetail3);
+    when(cloudStorageService.getBucketFileList("MockBucketName", "notebook")).thenReturn(blobDetailList);
+
+    // Will return 1 entry as only python files are filtered out
+    List<FileDetail> result = workspacesController.getNoteBookList("mockProjectName", "mockWorkspaceName").getBody();
+    assertEquals(result.size(), 1);
+
+    try {
+      result = workspacesController.getNoteBookList("mockProject", "mockWorkspace").getBody();
```

Comment:
nit: no reason to assign the result to a local variable or to call getBody() as the result is unused.